### ISSUE TITLE
[MySQL] Materialized views error

### ISSUE THREAD

**Issue by sorix6** [dimitri/pgloader#932]
- [x] pgloader --version
  
    ```
    pgloader version "3.6.1"
    compiled with SBCL 1.4.5.debian
    ```
  
  - [x] did you test a fresh compile from the source tree?
    
  - [x] did you search for other similar issues?
    I think that *[MySQL] MATERIALIZED VIEWS not working anymore after version 3.5.2 #918* references the same issue

  - [x] how can I reproduce the bug?

    Sorry, cannot include a dump

```
    LOAD DATABASE
        FROM mysql://user:pass@host:port/dbMySQL
        INTO postgresql://user:pass@host:port/dbPostgres

    WITH include no drop, create tables, create no indexes,
      workers = 8, concurrency = 1,
      multiple readers per thread, rows per range = 50000

   SET PostgreSQL PARAMETERS
      maintenance_work_mem to '128MB',
      work_mem to '12MB',
      search_path to 'dbPostgres, public, "$user"'

   SET MySQL PARAMETERS
      net_read_timeout  = '120',
      net_write_timeout = '120'

   MATERIALIZE VIEWS
        'matView' AS $$ select tm.active, tm.last_modified, tmi.name from tm as tm LEFT JOIN tmi as tmi ON tm.id = tmi.id_global; $$

   INCLUDING ONLY TABLE NAMES MATCHING 'matView'

   BEFORE LOAD DO
   $$ DROP TABLE IF EXISTS dbMySQL.matView; $$,
   $$ DROP TABLE IF EXISTS public.matView; $$,
   $$ create schema if not exists dbMySQL; $$,
   $$ alter database capture set search_path to dbMySQL, public; $$

   AFTER LOAD DO
   $$ ALTER TABLE dbMySQL.matView SET SCHEMA public $$,
   $$ DROP SCHEMA IF EXISTS dbMySQL $$
  
  set work_mem to '512 MB', maintenance_work_mem to '512 MB'
;

```

  - [x] pgloader output you obtain
  
```
ERROR mysql: MySQL Error [1064]: "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(NIL . matView...
```
  
The error is fixed by reverting the changes applied by the ```commit b8e8cf7d183d6beb81b84fc70c34834ac279a047``` on the file /src/parsers/command-materialize-views.lisp. But this change makes MSSQL views stop working.

**Comment by dimitri** [comment]
Thanks @sorix6 for another great bug report! Time to consider hacking your way in pgloader maybe? ;-)

**Comment by sorix6** [comment]
Hi @dimitri, thanks for the fix. 
Unfortunately, it does not seem to be working. The error generated by the schema is indeed gone, but the views no longer get copied into the target database.

**Comment by sorix6** [comment]
Hi @dimitri, thank you for the fix.

Unfortunately, it does not seem to be working. The error generated by the schema is indeed gone, but the views no longer get copied into the target database.

From my understanding, the problem comes from this query:

````
SQL MySQL: sending query: -- params: db-name
--         table-type-name
--         only-tables
--         only-tables
--         including
--         filter-list-to-where-clause incuding
--         excluding
--         filter-list-to-where-clause excluding
  select c.table_name, t.table_comment,
         c.column_name, c.column_comment,
         c.data_type, c.column_type, c.column_default,
         c.is_nullable, c.extra
    from information_schema.columns c
         join information_schema.tables t using(table_schema, table_name)
   where c.table_schema = 'mySQL' and t.table_type = 'VIEW'
         and table_name in ('(NIL . matView)')         
order by table_name, ordinal_position;
````

Commenting out the line
````
~:[~*~;and table_name in (~{'~a'~^,~})~]
````
in the file ````/src/sources/mysql/sql/list-all-columns.sql```` fixes the problem because we remove the filters on the views. The side effects are obvious, but I wasn't capable of fixing this with my current Lisp skills... :/

**Comment by dimitri** [comment]
Thanks for the detailed bug report, I think the attached patch fixes it this time. I forgot one place to edit. Once more I'm bitten by what I like to name _anonymous data structures_, where it's then hard to follow along what the code expects to see really. I should tighten all the things...
